### ISSUE TITLE
Fix 'cannot parse selector' error for nested HTML elements

### ISSUE THREAD

**Issue by daviddelusenet** [stylelint/stylelint#4401]
> Clearly describe the bug

I'm using `styled-components` in combination with `stylelint`. I first was using the the `stylelint-processor-styled-components` processor but I've now removed it since I found out `stylelint` support CSS-in-JS out of the box nowadays.

However, I'm now encountering a weird bug. I've the following code:

```css
export const VisibilitySwitch = styled.button`
    /* Some CSS here */

    ${({ theme, variant }) => variant === INPUT_PASSWORD_VARIANTS.COMPACT && css`
        /* Some CSS here */
    `}

    ${({ theme, variant }) => variant === INPUT_PASSWORD_VARIANTS.OUTLINE && css`
        /* Some CSS here */
    `}

    ${({ isDisabled, theme }) => isDisabled && css`
        /* Some CSS here */
    `}

    span {
        display: block;
    }
`;
```

This is all completely valid but because my `VisibilitySwitch` contains the nested `span` HTML element `stylelint` begins to throw the following errors:

```SHELL
src/app/components/organisms/InputPassword/InputPassword.sc.js
 23:5  ✖  Cannot parse selector              parseError
 23:5  ✖  Cannot parse selector              parseError
 23:5  ✖  Cannot parse selector              parseError
 23:5  ✖  Cannot parse selector              parseError
 24:9  ✖  Expected indentation of 4 spaces   indentation
 25:9  ✖  Expected indentation of 4 spaces   indentation
 26:9  ✖  Expected indentation of 4 spaces   indentation
 30:9  ✖  Expected indentation of 4 spaces   indentation
 31:9  ✖  Expected indentation of 4 spaces   indentation
 32:9  ✖  Expected indentation of 4 spaces   indentation
 36:9  ✖  Expected indentation of 4 spaces   indentation
 37:9  ✖  Expected indentation of 4 spaces   indentation
```
If I remove the `span` part everything works fine. The same errors get reported by my editor, following are some screenshots:

![Screenshot 2019-11-07 at 11 30 25](https://user-images.githubusercontent.com/1398418/68381449-091c0a00-0152-11ea-91c3-4913262a6b25.png)
![Screenshot 2019-11-07 at 11 30 37](https://user-images.githubusercontent.com/1398418/68381451-0a4d3700-0152-11ea-9d11-8f45faa1e47a.png)

This is not only happening in this component but in every styled component in my project which contains a nested HTML element declaration.

> Which rule, if any, is the bug related to?

`indentation` and `parseError`.

> What stylelint configuration is needed to reproduce the bug?

```
module.exports = {
    extends: [
        'stylelint-config-recommended',
        'stylelint-config-styled-components',
        'stylelint-config-concentric-order',
    ],
    rules: {
        indentation: 4,
        'length-zero-no-unit': true,
    },
};
```
> Which version of stylelint are you using?

`11.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint \"src/**/*.js\"` and inside of VS Code using [vscode-stylelint](https://github.com/shinnn/vscode-stylelint)

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Styled Components

> What did you expect to happen?

That no errors/warnings are triggered.

> What actually happened (e.g. what warnings or errors did you get)?

See above

**Comment by daviddelusenet** [comment]
I just found out two ways to prevent this behaviour, still not ideal, but at least I don't get errors anymore.

Fix one:

```css
export const VisibilitySwitch = styled.button`
    /* Some CSS here */

    ${({ theme, variant }) => variant === INPUT_PASSWORD_VARIANTS.COMPACT && css`
        /* Some CSS here */
    `}

    ${({ theme, variant }) => variant === INPUT_PASSWORD_VARIANTS.OUTLINE && css`
        /* Some CSS here */
    `}

    ${({ isDisabled, theme }) => isDisabled && css`
        /* Some CSS here */
    `}

    &:active {
        background: red;
    }

    span {
        display: block;
    }
`;
```

Adding something like `&:hover` or &:active` in between will solve the errors. This of course isn't a real solution since not all components need those selectors. However, I also discovered the following solution:

```css
export const VisibilitySwitch = styled.button`
    /* Some CSS here */

    span {
        display: block;
    }

    ${({ theme, variant }) => variant === INPUT_PASSWORD_VARIANTS.COMPACT && css`
        /* Some CSS here */
    `}

    ${({ theme, variant }) => variant === INPUT_PASSWORD_VARIANTS.OUTLINE && css`
        /* Some CSS here */
    `}

    ${({ isDisabled, theme }) => isDisabled && css`
        /* Some CSS here */
    `}
`;
```

Just putting the HTML element (the `span` in this case) above the modifiers 😅

**Comment by hudochenkov** [comment]
Looks like something wrong with the parser.

**Comment by hudochenkov** [comment]
It might be related to incorrect parsing of nested template literals https://github.com/stylelint/stylelint/issues/4119

**Comment by jeddy3** [comment]
We've forked the parser we use for CSS-in-JS [into stylelint organisation](https://github.com/stylelint/postcss-css-in-js), and we're looking for help to fix our fork to close this issue.

See https://github.com/stylelint/stylelint/issues/4574.
